### PR TITLE
test: validity for gate, job_type, measure, noise, pauli_string and other_instructions 

### PR DESCRIPTION
**Pull request by JulienCalistoTD**

### CODE REVIEW

File: mpqp/tools/circuit.py [hunk 1]
```diff
@@ -159,6 +171,53 @@ def random_instruction(
...
+    """
+    rng = np.random.default_rng(seed)
+
+    if noise_model is None:
+        noise_model = NOISE_MODELS
+
+    noise = rng.choice(np.array(noise_model))
+
+    if issubclass(noise, AmplitudeDamping):
+        prob = rng.uniform(0, 1)
```
Comment (Henri-ColibrITD) [inline mpqp/tools/circuit.py:203]: I'm not in love with this, I guess this is the reason why you asked if we couldn't have a `proba` attribute for all noises

File: tests/execution/test_validity.py [hunk 2]
```diff
@@ -292,3 +323,163 @@ def test_observable_ideal_case(
...
+        elif language in [Language.MY_QLM] and not isinstance(
+            noise_build, (Depolarizing, PhaseDamping)
+        ):
+            with pytest.raises(NotImplementedError):
+                noise_build.to_other_language(language)
+        else:
+            assert noise_build.to_other_language(language) is not None
+
+
+other_instr = [
```
Comment (Henri-ColibrITD) [inline tests/execution/test_validity.py:468]: I would have that as a fixture

File: tests/execution/test_validity.py [hunk 3]
```diff
@@ -292,3 +323,163 @@ def test_observable_ideal_case(
...
+            )  # pyright: ignore[reportUnnecessaryIsInstance]
+            and not isinstance(measure.basis, ComputationalBasis)
+        ):
+            with pytest.raises(NotImplementedError):
+                measure.to_other_language(language)
+        else:
+            assert measure.to_other_language(language) is not None
+
+
+pauli_strings = [Ip @ Xp @ Yp @ Zp, Xp + Zp, Yp]
```
Comment (Henri-ColibrITD) [inline tests/execution/test_validity.py:442]: fixture

**Review by Henri-ColibrITD — CHANGES_REQUESTED** [review]
tiny tweaks

File: tests/execution/test_validity.py (hunk 2 above)
Comment (JulienCalistoTD) [reply tests/execution/test_validity.py:468]: done

File: tests/execution/test_validity.py (hunk 3 above)
Comment (JulienCalistoTD) [reply tests/execution/test_validity.py:442]: done

**Pull request merged**